### PR TITLE
SSCS_WA_1_0 Release/feature branch

### PR DESCRIPTION
**Pull request by abul-syed**
### JIRA link (if applicable) ###

- [AM-2762](https://tools.hmcts.net/jira/browse/AM-2762) - SSCS_WA - Add mapping rules in ORM for STAFF users
- [AM-2763](https://tools.hmcts.net/jira/browse/AM-2763) - SSCS_WA - Add mapping rules in ORM for SSCS judges
- [AM-2764](https://tools.hmcts.net/jira/browse/AM-2764) - SSCS_WA - Add CASE role validation drool rules in RAS application for STAFF And JUDICIARY
- [AM-2765](https://tools.hmcts.net/jira/browse/AM-2765) - SSCS_WA - Enable Specific and Challenged access for SSCS for both STAFF and JUDICIARY (RAS)

plus:
- [AM-2858](https://tools.hmcts.net/jira/browse/AM-2858) - SSCS_WA - Final round of review and testing of combined feature branches for release SSCS_WA_1_0

### Change description ###

SSCS_WA_1_0 Release/feature branch:

- #1911
- #1912
- #1887
- #1888

Patch PRs:
- #1930
- #1934

> NB: This PR is has a dependency on hmcts/am-org-role-mapping-service#1496

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/roleconfig/role_common.json [hunk 1]
```diff
@@ -2553,5 +2697,71 @@
...
+          "values": ["STANDARD"]
+        },
+        "classification": {
+          "mandatory": true,
+          "values": ["PUBLIC"]
+        },
+        "attributes": {
+          "jurisdiction": {
+            "mandatory": true,
+            "values": ["EMPLOYMENT","SSCS"]
```
Comment (mattnayler) [inline src/main/resources/roleconfig/role_common.json:2725]: `"regional-centre-team-leader"` needs a unit test in AllServicesOrgRoleTest.java/shouldApproveRequestedRoleForOrg

File: src/main/resources/roleconfig/role_common.json [hunk 2]
```diff
@@ -2553,5 +2697,71 @@
...
```
Comment (mattnayler) [inline src/main/resources/roleconfig/role_common.json:2758]: `"regional-centre-admin"` needs a unit test in AllServicesOrgRoleTest.java/shouldApproveRequestedRoleForOrg

**Review by mattnayler — CHANGES_REQUESTED** [review]
`role_common.json` line 600
> tribunal-caseworker role of `"type": "CASE"` should not be configured for SSCS.  (i.e. only the `"type": "ORGANISATION"` is valid)

File: src/main/resources/roleconfig/role_common.json [hunk 3]
```diff
@@ -557,7 +557,7 @@
         "attributes": {
           "jurisdiction": {
             "mandatory": true,
-            "values": ["IA","CIVIL","PRIVATELAW", "PUBLICLAW", "EMPLOYMENT"]
+            "values": ["IA","CIVIL","PRIVATELAW", "PUBLICLAW", "EMPLOYMENT","SSCS"]
```
[the remark this answers is not in the document]
Comment (abul-syed) [reply src/main/resources/roleconfig/role_common.json:560]: tribunal-caseworker 'case' role removed for sscs & leadership-judge added in unit test. see 777f518

File: src/main/resources/roleconfig/role_common.json (hunk 1 above)
Comment (abul-syed) [reply src/main/resources/roleconfig/role_common.json:2725]: added in 777f518

File: src/main/resources/roleconfig/role_common.json (hunk 2 above)
Comment (abul-syed) [reply src/main/resources/roleconfig/role_common.json:2758]: added in 777f518

**Review by mattnayler — APPROVED** [review]
Approved after merger of #1930, 49aa3fef6dbf79a11eebe2fe8a36a0e5e5fb73fc.

File: src/main/resources/roleconfig/role_sscs.json [hunk 4]
```diff
@@ -295,10 +333,10 @@
     ]
   },
   {
```
Comment (mikebrownccd) [inline src/main/resources/roleconfig/role_sscs.json:335]: "superuser" is removed, is this correct?

File: src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl [hunk 5]
```diff
@@ -19,49 +19,94 @@ import java.time.ZonedDateTime;
...
 
 /*
- * SSCS "tribunal-caseworker" case validation rule.
+ * SSCS "registrar" case validation rule.
  */
 
-rule "sscs_tribunal_caseworker_create_case_roles"
+rule "sscs_registrar_create_case_roles"
 when
     $f:  FeatureFlag(status && flagName == FeatureFlagEnum.SSCS_WA_1_0.getValue())
```
Comment (mikebrownccd) [inline src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl:27]: Strange that it seems we already have FeatureFlagEnum.SSCS_WA_1_0 in the codebase before the changes for this branch. Any idea why?

File: src/main/resources/roleconfig/role_sscs.json (hunk 4 above)
Comment (abul-syed) [reply src/main/resources/roleconfig/role_sscs.json:335]: yeah. `superuser` was in the old SSCS config. but its been removed from the SSCS config page by the BA

File: src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl (hunk 5 above)
Comment (abul-syed) [reply src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl:27]: the SSCS config was initially done by the previous devs a year ago. back then, they used flags. maybe they can be retired?

Comment (mikebrownccd) [reply src/main/resources/validationrules/sscs/sscs-case-role-validation-legal-ops.drl:27]: Matt is aware and happy with this reuse. resolved

**Review by mikebrownccd — APPROVED** [review]
LGTM

**Pull request merged**